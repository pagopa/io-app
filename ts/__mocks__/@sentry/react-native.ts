export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
