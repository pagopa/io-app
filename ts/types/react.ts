import { Component, ComponentType, StatelessComponent } from "react";
import {
  NavigationInjectedProps,
  NavigationScreenProps
} from "react-navigation";

/**
 * Evaluates to the Props type of a React component
 */
export type ComponentProps<C> = C extends StatelessComponent<infer P1>
  ? P1
  : C extends Component<infer P2>
    ? P2
    : C extends ComponentType<infer P3> ? P3 : never;

/**
 * Infers the type of the navigation params of a component
 */
export type InferNavigationParams<
  C,
  P = ComponentProps<C>
> = P extends NavigationInjectedProps<infer N>
  ? N
  : P extends NavigationScreenProps<infer N1> ? N1 : never;
