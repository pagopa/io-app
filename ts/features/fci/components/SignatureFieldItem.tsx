import * as React from "react";
import { View, StyleSheet } from "react-native";
import { IOColors, Icon } from "@pagopa/io-app-design-system";
import { H4 } from "../../../components/core/typography/H4";
import { IOStyles } from "../../../components/core/variables/IOStyles";
import TouchableDefaultOpacity from "../../../components/TouchableDefaultOpacity";
import { Link } from "../../../components/core/typography/Link";
import I18n from "../../../i18n";

type Props = {
  title: string;
  value?: boolean;
  disabled?: boolean;
  onChange: (_: boolean) => void;
  onPressDetail: () => void;
};

const styles = StyleSheet.create({
  container: {
    paddingTop: 16,
    paddingBottom: 8,
    marginBottom: 16,
    flexDirection: "column",
    borderBottomColor: IOColors.greyLight,
    borderBottomWidth: 1
  },
  details: {
    paddingTop: 16,
    paddingBottom: 8
  },
  titleMargin: { marginRight: 22, flex: 1 }
});

const SignatureFieldItem = (props: Props) => {
  const [checked, setChecked] = React.useState(props.value || false);
  const onChange = (value: boolean) => {
    setChecked(value);
    props.onChange(value);
  };

  return (
    <View style={styles.container}>
      <View style={IOStyles.row}>
        <H4 style={styles.titleMargin} testID="SignatureFieldItemTitleTestID">
          {props.title}
        </H4>
        <TouchableDefaultOpacity
          accessibilityRole={"checkbox"}
          accessibilityLabel={
            checked
              ? I18n.t("features.fci.signatureFields.accessibility.selected")
              : I18n.t("features.fci.signatureFields.accessibility.unselected")
          }
          accessibilityState={{ selected: checked }}
          testID={"SignatureFieldItemButtonTestID"}
          onPress={() => {
            onChange(!checked);
          }}
          disabled={props.disabled}
          style={{ alignSelf: "center" }}
        >
          <Icon
            testID="SignatureFieldItemCheckboxTestID"
            name={checked ? "legCheckOn" : "legCheckOff"}
            color={
              checked && !props.disabled
                ? "blue"
                : props.disabled
                ? "grey"
                : "bluegreyDark"
            }
            size={24}
          />
        </TouchableDefaultOpacity>
      </View>
      <View style={[IOStyles.row, styles.details]}>
        <Link
          accessibilityLabel={I18n.t(
            "features.fci.signatureFields.showOnDocument"
          )}
          accessibilityRole="link"
          accessibilityHint={I18n.t(
            "features.fci.signatureFields.accessibility.fieldDetailHint"
          )}
          testID="SignatureFieldItemDetailTestID"
          onPress={props.onPressDetail}
        >
          {I18n.t("features.fci.signatureFields.showOnDocument")}
        </Link>
      </View>
    </View>
  );
};

export default SignatureFieldItem;
