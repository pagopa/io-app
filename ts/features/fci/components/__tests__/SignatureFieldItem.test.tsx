import * as React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import SignatureFieldItem from "../SignatureFieldItem";
import I18n from "../../../../i18n";

type Props = {
  title: string;
  value?: boolean;
  onChange: (_: boolean) => void;
  onPressDetail: () => void;
};

describe("Test SignatureFieldItem component", () => {
  it("should render a SignatureFieldItem component with props correctly", () => {
    const props = {
      title: "Clause title 1",
      onChange: jest.fn(),
      onPressDetail: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
  it("should render a SignatureFieldItem component with right title 'Clause title 1'", () => {
    const props = {
      title: "Clause title 1",
      onChange: jest.fn(),
      onPressDetail: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    expect(component.queryByText("Clause title 1")).toBeTruthy();
  });
  it("should render a SignatureFieldItem component with checkbox unchecked", () => {
    const props = {
      title: "Clause title 1",
      onChange: jest.fn(),
      onPressDetail: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    expect(
      component.getByTestId("SignatureFieldItemCheckboxTestID")
    ).toBeTruthy();
    expect(component.queryAllByText("io-checkbox-off")).toBeTruthy();
  });
  it("should render a SignatureFieldItem component with checkbox checked", () => {
    const props = {
      title: "Clause title 1",
      value: true,
      onChange: jest.fn(),
      onPressDetail: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    expect(
      component.getByTestId("SignatureFieldItemCheckboxTestID")
    ).toBeTruthy();
    expect(component.queryAllByText("io-checkbox-on")).toBeTruthy();
  });
  it("should render a SignatureFieldItem component with checkbox clickable", () => {
    const onPress = jest.fn();
    const props = {
      title: "Clause title 1",
      value: true,
      onChange: onPress,
      onPressDetail: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    const rightButton = component.getByTestId("SignatureFieldItemButtonTestID");
    expect(rightButton).toBeTruthy();
    expect(rightButton).toBeEnabled();
    fireEvent.press(rightButton);
    expect(component.queryAllByText("io-checkbox-on")).toBeTruthy();
    fireEvent.press(rightButton);
    expect(component.queryAllByText("io-checkbox-off")).toBeTruthy();
    expect(onPress).toHaveBeenCalledTimes(2);
  });
  it("should render a SignatureFieldItem component with right text for details link", () => {
    const props = {
      title: "Clause title 1",
      value: true,
      onChange: jest.fn(),
      onPressDetail: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    const detailsLink = component.queryAllByText(
      I18n.t("features.fci.signatureFields.showOnDocument")
    );
    expect(detailsLink).toBeTruthy();
  });
  it("should render a SignatureFieldItem component with details link clickable", () => {
    const onPress = jest.fn();
    const props = {
      title: "Clause title 1",
      value: true,
      onChange: jest.fn(),
      onPressDetail: onPress
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    const detailsLink = component.getByTestId("SignatureFieldItemDetailTestID");
    expect(detailsLink).toBeTruthy();
    fireEvent.press(detailsLink);
    fireEvent.press(detailsLink);
    expect(onPress).toHaveBeenCalledTimes(2);
  });
});

const renderComponent = (props: Props) =>
  render(<SignatureFieldItem {...props} />);
