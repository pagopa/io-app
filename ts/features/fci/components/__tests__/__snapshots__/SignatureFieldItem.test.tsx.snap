// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SignatureFieldItem component should render a SignatureFieldItem component with props correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#E6E9F2",
      "borderBottomWidth": 1,
      "flexDirection": "column",
      "marginBottom": 16,
      "paddingBottom": 8,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      color="bluegreyDark"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 16,
          "lineHeight": 22,
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "marginRight": 22,
          },
          Object {
            "fontSize": 16,
            "lineHeight": 22,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "700",
          },
        ]
      }
      testID="SignatureFieldItemTitleTestID"
      weight="Bold"
      weightColorFactory={[Function]}
    >
      Clause title 1
    </Text>
    <View
      accessibilityLabel="Field not selected"
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "selected": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
      testID="SignatureFieldItemButtonTestID"
    >
      <RNSVGSvgView
        accessibilityElementsHidden={true}
        accessibilityLabel=""
        accessible={false}
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        color={4279710285}
        focusable={false}
        height={24}
        importantForAccessibility="no-hide-descendants"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "color": "#17324D",
            },
            Object {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        testID="SignatureFieldItemCheckboxTestID"
        tintColor={4279710285}
        vbHeight={22}
        vbWidth={22}
        width={24}
      >
        <RNSVGGroup>
          <RNSVGPath
            clipRule={0}
            d="M19.25 0H2.75C1.23122 0 0 1.23122 0 2.75v16.5C0 20.7688 1.23122 22 2.75 22h16.5c1.5188 0 2.75-1.2312 2.75-2.75V2.75C22 1.23122 20.7688 0 19.25 0ZM1.83333 2.75c0-.50626.41041-.91667.91667-.91667h16.5c.5063 0 .9167.41041.9167.91667v16.5c0 .5063-.4104.9167-.9167.9167H2.75c-.50626 0-.91667-.4104-.91667-.9167V2.75Z"
            fill={
              Array [
                2,
              ]
            }
            fillRule={0}
            propList={
              Array [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "paddingBottom": 8,
          "paddingTop": 16,
        },
      ]
    }
  >
    <Text
      accessibilityHint="Go to the page that displays the signature field"
      accessibilityLabel="See on document"
      accessibilityRole="link"
      color="blue"
      defaultColor="blue"
      defaultWeight="SemiBold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 16,
          "textDecorationLine": "underline",
        }
      }
      onPress={[MockFunction]}
      style={
        Array [
          Object {
            "fontSize": 16,
            "textDecorationLine": "underline",
          },
          Object {
            "color": "#0073E6",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      testID="SignatureFieldItemDetailTestID"
      weight="SemiBold"
    >
      See on document
    </Text>
  </View>
</View>
`;
